openedx/core/lib/django_courseware_routers.py

```diff
@@ -45,13 +45,15 @@ def allow_relation(self, obj1, obj2, **hints):  # pylint: disable=unused-argumen
             return False
         return None
 
-    def allow_migrate(self, db, model):  # pylint: disable=unused-argument
+    def allow_migrate(self, db, app_label, model_name=None, **hints):  # pylint: disable=unused-argument
         """
         Only sync StudentModuleHistoryExtended to StudentModuleHistoryExtendedRouter.DATABASE_Name
         """
-        if self._is_csmh(model):
-            return db == self.DATABASE_NAME
-        elif db == self.DATABASE_NAME:
+        if model_name is not None:
+            model = hints.get('model')
```

Comment:
> if model_name is non-None, then hints has a model key with the model in it. RunSQL (and RunPython) calls pass in None for model_name.

@nedbat At least in our version of Django (1.8.17) I'm having trouble finding an indication that this is true.

https://github.com/django/django/blob/d3d12fc11da56e4ea8af37a22a9a0aa6579ab2d5/django/db/migrations/migration.py#L112
https://github.com/django/django/blob/d3d12fc11da56e4ea8af37a22a9a0aa6579ab2d5/django/db/migrations/operations/special.py#L178

It looks like this change is causing some migrations to fail for fresh builds on OpenCraft sandboxes for `contenttypes.0002_remove_content_type_name`, where the stack trace indicates that model_name is getting populated positionally, but `hints.get('model')` is returning `None`:

```
2017-02-13 19:26:40,719 INFO 4054 [dd.dogapi] dog_stats_api.py:66 - Initializing dog api to use statsd: localhost, 8125
Traceback (most recent call last):
  File 
"manage.py
", line 116, in <module>
    execute_from_command_line([sys.argv[0]] + django_args)
  File 
"/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/management/__init__.py
", line 354, in execute_from_command_line
    utility.execute()
  File 
"/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/management/__init__.py
", line 346, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File 
"/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/management/base.py
", line 394, in run_from_argv
    self.execute(*args, **cmd_options)
  File 
"/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/management/base.py
", line 445, in execute
    output = self.handle(*args, **options)
  File 
"/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/management/commands/migrate.py
", line 222, in handle
    executor.migrate(targets, plan, fake=fake, fake_initial=fake_initial)
  File 
"/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/db/migrations/executor.py
", line 110, in migrate
    self.apply_migration(states[migration], migration, fake=fake, fake_initial=fake_initial)
  File 
"/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/db/migrations/executor.py
", line 148, in apply_migration
    state = migration.apply(state, schema_editor)
  File 
"/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/db/migrations/migration.py
", line 112, in apply
    operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
  File 
"/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/db/migrations/operations/special.py
", line 178, in database_forwards
    if router.allow_migrate(schema_editor.connection.alias, app_label, **self.hints):
  File 
"/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/db/utils.py
", line 361, in allow_migrate
    allow = method(db, app_label, **hints)
  File 
"/edx/app/edxapp/edx-platform/openedx/core/lib/django_courseware_routers.py
", line 54, in allow_migrate
    if self._is_csmh(model):
  File 
"/edx/app/edxapp/edx-platform/openedx/core/lib/django_courseware_routers.py
", line 18, in _is_csmh
    model._meta.app_label == 'coursewarehistoryextended' and  # pylint: disable=protected-access
AttributeError: 'NoneType' object has no attribute '_meta'", "stdout": "Operations to perform:
  Synchronize unmigrated apps: eventtracking_django, djcelery, release_util, statici18n, django_sites_extensions, fileupload, service_status, coursegraph, method_override, edxmako, openassessment, datadog, common_views, block_structure, rules, maintenance, provider, staticfiles, edx_jsme, monitoring, pipeline, simple_history, require, messages, config_models, xblock, django_nose, edx_sga
```